src/Exceptions/ServiceException.php

```diff
@@ -3,52 +3,78 @@
 namespace Honeybadger\Exceptions;
 
 use Exception;
+use Throwable;
 
 class ServiceException extends Exception
 {
     /**
-     * @return \Honeybadger\Exceptions\ServiceException
+     * @return ServiceException
      */
     public static function invalidApiKey(): self
     {
         return new static('The API key provided is invalid.');
     }
 
     /**
-     * @return \Honeybadger\Exceptions\ServiceException
+     * @return ServiceException
      */
     public static function invalidPayload(): self
     {
         return new static('The payload sent to Honeybadger was invalid.');
     }
 
     /**
-     * @return \Honeybadger\Exceptions\ServiceException
+     * @return ServiceException
      */
     public static function rateLimit(): self
     {
         return new static('You have hit your exception rate limit.');
     }
 
     /**
-     * @return \Honeybadger\Exceptions\ServiceException
+     * @return ServiceException
      */
     public static function serverError(): self
     {
         return new static('There was an error on our end.');
     }
 
+    /**
+     * @param int $code
+     * @return ServiceException
+     */
     public static function unexpectedResponseCode(int $code): self
     {
         return new static("Unexpected HTTP response code: $code");
     }
 
-    public static function generic(\Throwable $e = null): self
+    /**
+     * @param Throwable|null $e
+     * @return self
+     */
+    public static function generic(Throwable $e = null): self
```

Comment:
Small note: doc blocks such as these are an artifact of older PHP, when parameter types weren't available. This docblock essentially repeats information you can see clearly in the method definition. Your IDE may be bugging you to add doc blocks for all methods for consistency, but in some PHP circles, a more pragmatic approach is favoured: only add such doc blocks when they add extra information (eg additional information about the parameters or the method). That way we, we cut down on visual noise.

Reply:
Indeed, the IDE is suggesting all sorts of things and that's probably how I ended with these.

Reply:
Should I keep the doc blocks where the return is an array of a specific type? For example, the doc block below provide information about the type of the items in the array.
```
    /**
     * @param array $checkins
     * @return Checkin[]
     */
    public function sync(array $checkins): array
```

Reply:
Ah, yeah, that one provides more value, since PHP doesn't have generics yet, so you can keep it if you want.

There's a setting to disable the inspection that always suggests docblocks, don't remember the name.